backend/src/bundles/user-details/helpers/apply-filters/helpers/apply-type-search-filter.ts

```diff
@@ -0,0 +1,16 @@
+import { type QueryBuilder } from 'objection';
+
+import { SearchType } from '../../../enums/enums.js';
+import { type UserDetailsSearchUsersRequestDto } from '../../../types/types.js';
+import { type UserDetailsModel } from '../../../user-details.model.js';
+
+const applyTypeSearchFilter = (
+    builder: QueryBuilder<UserDetailsModel, UserDetailsModel[]>,
+    payload: UserDetailsSearchUsersRequestDto,
+): void => {
+    if (payload.searchType && payload.searchType === SearchType.ACTIVE) {
+        void builder.where('searchType', payload.searchType);
+    }
+};
```

Comment:
In this helper you use only payload.searchType, so you don't need to provide all payload to this helper